lib/sequenceserver/blast/job.rb

```diff
@@ -6,21 +6,32 @@ module BLAST
     # Extends SequenceServer::Job to describe a BLAST job.
     class Job < Job
       def initialize(params)
-        validate params
-        super do
-          @method    = params[:method]
-          @qfile     = store('query.fa', params[:sequence])
-          @databases = Database[params[:databases]]
-          @options   = params[:advanced].to_s.strip + defaults
-          @advanced_params = parse_advanced params[:advanced]
-        end
-      end
+        
+        if params.key?(:xml)
+	  super do	
+	    @xml_import = params[:xml]
+	    @advanced_params = {}
+	    @databases = []
+	    done!
+	  end   
+ 
+	else     
+	  validate params
+          super do
+            @method    = params[:method]
+            @qfile     = store('query.fa', params[:sequence])
+            @databases = Database[params[:databases]]
+            @options   = params[:advanced].to_s.strip + defaults
+            @advanced_params = parse_advanced params[:advanced]
+	  end 	
+	end
+      end 
 
       attr_reader :advanced_params
 
       # :nodoc:
       # Attributes used by us - should be considered private.
-      attr_reader :method, :qfile, :databases, :options
+      attr_reader :method, :qfile, :databases, :options, :xml_import
```

Comment:
Please move the new attr_reader to a new line, document it (that there are two modes - generated XML and imported XML), and call it `imported_xml_file`.